scipy/stats/tests/test_axis_nan_policy.py

```diff
@@ -803,10 +804,15 @@ def test_other_axis_tuples(axis):
     np.testing.assert_array_equal(res, res2)
 
 
-def test_gmean_mixed_mask_nan_weights():
+@pytest.mark.parametrize(("weighted_fun_name"), ["gmean", "hmean"])
+def test_gmean_mixed_mask_nan_weights(weighted_fun_name):
     # targeted test of _axis_nan_policy_factory with 2D masked sample:
     # omitting samples with masks and nan_policy='omit' are equivalent
     # also checks paired-sample sentinel value removal
+
+    weighted_fun = getattr(stats, weighted_fun_name)
+    weighted_fun_ma = getattr(stats.mstats, weighted_fun_name)
+
     m, n = 3, 20
     axis = -1
 
```

Comment:
Optional. Since we are at it, can you update the usages of `np.random.rand` bellow?

Reply:
Does minimum supported numpy `Generator` have `rand`? I'm assuming not, so I used `uniform`. This is one of the reasons I don't typically do this, especially in tests where we just need numbers.

Reply:
No there is no `rand` on `Generator`, the replacement is `random`. `uniform` is fine as well.